### PR TITLE
Use controllerutil.AddFinalizer

### PR DESCRIPTION
**Pull request by cuppett**
**Description of the change:**

Changes addFinalizer explicit code to controllerutil.AddFinalizer 

**Motivation for the change:**

To stay consistent with the example (RemoveFinalizer called just above)

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [-] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [+] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by jmrodri** [comment]
@cuppett can you add the signoff to your commit. Not sure what enabled the DCO check but it seems across the org. To do that just run this on your branch then force push up your change:

```
# this will add the signoff to your commit
git rebase HEAD~1 --signoff
# this will repush the changes back to the PR
git push --force origin cuppett-patch-1
```

File: website/content/en/docs/building-operators/golang/advanced-topics.md [hunk 1]
```diff
@@ -183,9 +183,8 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
     // Add finalizer for this CR
     if !contains(memcached.GetFinalizers(), memcachedFinalizer) {
-        if err := r.addFinalizer(reqLogger, memcached); err != nil {
-            return ctrl.Result{}, err
-        }
+        controllerutil.AddFinalizer(memcached, memcachedFinalizer)
+        r.Update(context.TODO(), memcached)
```
Comment (jmrodri) [inline website/content/en/docs/building-operators/golang/advanced-topics.md:187]: `r.Update` returns an error. We'll want to check that. Similar to how we did it in `addFinalizer` method you removed.

```golang
    err := r.Update(context.TODO(), m)
    if err != nil {
        reqLogger.Error(err, "Failed to update Memcached with finalizer")
        return ctrl.Result{}, err
```

**Review by jmrodri — CHANGES_REQUESTED** [review]
Thanks for the PR. Two small items:
1) add an error check to the example
2) add signoff to the commits

You can add a signoff during your commit with `git commit --signoff` OR after the fact by using rebase `git rebase HEAD~N --signoff` where `N` is the number of commits you want to add the signoff to.

File: website/content/en/docs/building-operators/golang/advanced-topics.md (hunk 1 above)
Comment (cuppett) [reply website/content/en/docs/building-operators/golang/advanced-topics.md]: :+1: Done and also noticed controllerutil.ContainsFinalizer is a thing too.

**Review by estroz — APPROVED** [review]
@cuppett thanks!

/lgtm

File: website/content/en/docs/building-operators/golang/advanced-topics.md [hunk 2]
```diff
@@ -182,8 +182,10 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
     }
 
     // Add finalizer for this CR
-    if !contains(memcached.GetFinalizers(), memcachedFinalizer) {
-        if err := r.addFinalizer(reqLogger, memcached); err != nil {
+    if !controllerutil.ContainsFinalizer(memcached, memcachedFinalizer) {
+        controllerutil.AddFinalizer(memcached, memcachedFinalizer)
+        err = r.Update(context.TODO(), memcached)
```
Comment (estroz) [inline website/content/en/docs/building-operators/golang/advanced-topics.md:187]: Missed this in my review:
```suggestion
        err = r.Update(ctx, memcached)
```

Comment (cuppett) [reply website/content/en/docs/building-operators/golang/advanced-topics.md]: Done. Thanks and good catch!

**Review by jmrodri — APPROVED** [review]

**Pull request merged**